### PR TITLE
forwarder::pkg_provider should default to $splunk::params::pkg_provider

### PR DESCRIPTION
**Pull request by hunner**
Closes #21

### CODE REVIEW

**Comment by hunner** [comment]
This is backwards-incompatible

**Comment by hunner** [comment]
The base class defaults to using `$splunk::params::pkg_provider` as the chosen package provider but the `splunk::forwarder` define defaults to undef for the provider. This commit now syncronizes the behavior to be consistent between both items.

**Comment by hunner** [comment]
@asquelt @kyledecot Does this look appropriate to you? Would this be considered a "bug fix" because the behavior of the module was inconsistent? (re #21)

File: manifests/forwarder.pp [hunk 1]
```diff
@@ -54,7 +54,7 @@
   $splunkd_listen    = '127.0.0.1',
   $purge_inputs      = false,
   $purge_outputs     = false,
-  $pkg_provider      = undef,
+  $pkg_provider      = $splunk::pkg_provider,
```
Comment (Ziaunys) [inline manifests/forwarder.pp]: This should be $splunk::params::pkg_provider, right?

**Comment by discreet** [comment]
I've had to change the code to implement this fix. Will this be updated and merged soon? Seems like it just needs a very simple minor change to the PR.

**Comment by hunner** [comment]
Updated, and calling this a bugfix instead of backwards-incompatible feature as the intent (as shown in the base class) is to follow `splunk::params::pkg_provider` and this failed to follow the original intent.

**Pull request merged**